src/app/dashboard/components/dashboard-disease-menu/dashboard-disease-menu.component.ts

```diff
@@ -1,8 +1,48 @@
-import { Component } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { EntityLink } from '../../../_constants/common';
+import { Disease } from '../../../_constants/enums';
+import { Filter } from '../../../_models/common';
+import { EnumToKeyValuePipe } from '../../../_pipes/enum-to-key-value/enum-to-key-value.pipe';
+import { FilterService } from '../../../_services/filter.service';
 
 @Component({
   selector: 'app-dashboard-disease-menu',
   templateUrl: './dashboard-disease-menu.component.html',
   styleUrls: ['./dashboard-disease-menu.component.scss'],
 })
-export class DashboardDiseaseMenuComponent {}
+export class DashboardDiseaseMenuComponent implements OnInit, OnDestroy {
+  links: EntityLink[] = [];
+  pipe = new EnumToKeyValuePipe();
+  currentFilters: Filter[] = [];
+  diseases = this.pipe.transform(Disease);
+  subscriptions: Subscription[] = [];
+  constructor(public filterService: FilterService) {}
+
+  removePrevDiseaseFilter(): void {
+    const newFilters: Filter[] = [];
+    this.currentFilters.forEach((f: Filter) => {
+      if (f.field !== 'disease') {
+        newFilters.push(f);
+      }
+    });
+    this.currentFilters = newFilters;
```

Comment:
this could be simplified using Array.filter

Reply:
Resolved.